app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -114,26 +114,51 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 
 }
 
-    <% if @force -%>
-    yum remove -y katello-ca-consumer*
-    <% end -%>
-
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
-
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
-    else
-        yum localinstall $CONSUMER_RPM -y
-    fi
-
-    rm -f $CONSUMER_RPM
-
-    subscription-manager register <%= '--force' if @force %> --org='<%= @organization.label %>' --activationkey='<%= activation_keys %>' || <%= @ignore_subman_errors ? 'true' : 'exit 1' %>
-    register_katello_host | bash
+  <% if @force -%>
+  yum remove -y katello-ca-consumer*
+  <% end -%>
+
+  KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/<%= @url_host %>.pem
+  CFG=/etc/rhsm/rhsm.conf
+
+  if ! command -v subscription-manager &> /dev/null
+  then
+    yum install -y subscription-manager
+  fi
+
+  if ! [ -f $CFG ] ; then
+    echo "'$CFG' not found, cannot configure subscription-manager"
+    exit 1
+  fi
+
+  # Prepare SSL certificate
+  cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+  chmod 644 $KATELLO_SERVER_CA_CERT
+
+  if [ -d /etc/pki/ca-trust/source/anchors ]; then
+    update-ca-trust enable
+    cp -f $KATELLO_SERVER_CA_CERT /etc/pki/ca-trust/source/anchors
+    update-ca-trust
+  fi
+
+  # Configure subscription-manager
+  test -f $CFG.BAK || cp $CFG $CFG.BAK
+  subscription-manager config \
+    --server.hostname="<%= @url_host %>" \
+    --server.port="<%= @url_port %>" \
+    --server.prefix="/rhsm" \
+    --rhsm.repo_ca_cert="$KATELLO_SERVER_CA_CERT" \
+    --rhsm.baseurl="<%= @url %>pulp/content/"
```

Comment:
I don't think this is right. This looks like a Pulp 2 URL.

I think there should be a macro from Katello for the correct content URL (note that I'm not saying there is one today, but there should be).

Reply:
From this PRs [0][1] & redmine [2] looks like all content can be get from `/pulp/content`

[0] https://github.com/Katello/katello/pull/9261
[1] https://github.com/theforeman/puppet-foreman_proxy_content/commit/28819d0ae1cfea204a7fe2c80093d88d0f2918e2
[2] https://projects.theforeman.org/issues/31867

Reply:
Ah you're right. However, I still thing a macro is needed. My reasoning is that it should respect the Pulp content URL that Katello can discover via the Smart Proxy.

Today it's not supported, but it is my desire to support complex architectures where a Smart Proxy is on a different host than the Pulp instance. There is currently a working group looking at those advanced scenarios. That why I think a `pulp_content_url(host)` makes sense.

While we're at it, a `rhsm_url(host)` macro is also better than assuming it's on the same URL as the Smart Proxy.

And yes, I do believe Katello may be doing the same thing wrong right now. That's why I won't block on it, but I'm working on this on all fronts.